File: 프로그래머스/0/181859. 배열 만들기 6/배열 만들기 6.js
function solution(arr) {
    let stk = [];
    
    for (let i = 0; i < arr.length; i++) {
        if (stk.length === 0 || stk[stk.length - 1] !== arr[i]) {
            stk.push(arr[i]);
        } else {
            stk.pop();
        }
    }
    
    return stk.length > 0 ? stk : [-1];
}

File: 프로그래머스/0/181859. 배열 만들기 6/배열 만들기 6.java
import java.util.*;

class Solution {
    public int[] solution(int[] arr) {
        List<Integer> stk = new ArrayList<>();
        
        int i = 0;
        while (i < arr.length) {
            int size = stk.size();
            
            if (size < 1) {
                stk.add(arr[i]);
                i++;
            } else if (stk.get(size - 1) == arr[i]) {
                stk.remove(size - 1);
                i++;
            } else {
                stk.add(arr[i]);
                i++;
            }
        }
        
        if (stk.size() < 1) stk.add(-1);
        
        return stk.stream().mapToInt(Integer::intValue).toArray();
    }
}